File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/GetLocationActionParameters.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint parameters for a Get Location action.
    /// </summary>
    public class GetLocationActionParameters: ICardActionParameters
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="GetLocationActionParameters"/> class.
        /// </summary>
        public GetLocationActionParameters()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the choose location on map of type <see cref="bool"/>.
        /// </summary>
        /// <value>This value indicates whether a location on the map can be chosen.</value>
        [JsonProperty(PropertyName = "ChooseLocationOnMap")]
        public bool ChooseLocationOnMap { get; set; }
    }
}

File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/getLocationActionParameters.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

/**
 * Sharepoint GetLocationActionParameters object for get location action
 */
export class GetLocationActionParameters {
    private chooseLocationOnMap: boolean;
    
    /**
     * Initializes a new instance of the GetLocationActionParameters class
     */
    public GetLocationActionParameters() {
        // Do nothing
    }

    /**
     * Sets whether a location on the map can be chosen of type boolean
     */
    public set ChooseLocationOnMap(chooseLocationOnMap: boolean) {
        this.chooseLocationOnMap = chooseLocationOnMap;
    }

    /**
     * Gets whether a location on the map can be chosen of type boolean
     */
    public get ChooseLocationOnMap(): boolean {
        return this.chooseLocationOnMap;
    }
}